quint/src/parsing/ToIrListener.ts

```diff
@@ -430,6 +430,10 @@ export class ToIrListener implements QuintListener {
         // This shouldn't be visible to users
         const paramName = `__${fieldName}Param`
 
+        // def declarations and the lambdas they are associated with should share
+        // the same ID
```

Comment:
Do we really need this? I feel a bit insecure about this changes because right now, every IR component has a distinct ID, and it might be the case that we are assuming this for some things across the code base.

I actually like the idea of making this a general thing, but I'm not 100% sure we can.

Violating this (perhaps unimportant) assumption of distinct ids only in this new case that's not heavily tested yet seems a bit dangerous to me.

Reply:
I don't think we need this to fix any behavior, but i I introduced it because it is how, when I checked, the other declarations were working already, so i thought I was making making things more consistent! Sounds like we have a misunderstanding one one side or the other. Let me double check.

Reply:
As per our discussion, I've reverted those changes. Thank you for the careful review and catching this.

I'll also prepare a followup to fix this for other operators.